### PR TITLE
MGDAPI-3665 refactor getSupportedVersions

### PR DESCRIPTION
**Pull request by makslion**
## JIRA link
[MGDAPI-3665](https://issues.redhat.com/browse/MGDAPI-3665)

## What
Refactor getSupportedVersions to enable an accurate supported versions retrieval for RHOAM. This enables the usage of an index referenced in the addon.yaml. 

## Verification
1. Build CLI by `make build/cli`
2. Run `./delorean -m=3 --olmType=managed-api-service pipeline supported-versions` for RHOAM
3. Run `./delorean -m=3 --olmType=integreatly-operator pipeline supported-versions` to ensure RHMI still works

### CODE REVIEW

**Review by carlkyrillos — COMMENTED** [review]
Verified that the getSupportedVersions command works for both RHOAM and RHMI. Reviewed the changed files and added a few comments but overall looks good

File: cmd/getSupportedVersions.go [hunk 1]
```diff
@@ -134,6 +150,31 @@ func (c *getSupportedVersionsCmd) run(ctx context.Context) ([]string, error) {
...
+		IndexImage string `json:"indexImage"`
+	}
+
+	data := indexFiled{}
+	err := utils.PopulateObjectFromYAML(root, &data)
+	if err != nil {
+		return err
+	}
+	indexSha := data.IndexImage
+	// assuming opm is installed
```
[the remark this answers is not in the document]
Comment (makslion) [reply cmd/getSupportedVersions.go:167]: We also need docker for delorean. I've raised that question before and mentioned you on the thred :thinking: For me docker and opm should be in the `prerequisites` section

File: cmd/getSupportedVersions_test.go [hunk 2]
```diff
@@ -383,25 +383,22 @@ func TestGetOlmTypePath(t *testing.T) {
...
 		t.Run(c.description, func(t *testing.T) {
-			resultBundlePath, resultFilePath, err := getOlmTypePaths(c.olmType)
+			paths, err := getOlmTypePaths(c.olmType)
 			if err != nil {
 				if c.hasError && err.Error() != c.expectedError {
 					t.Fatalf("Did not get expected error. Expected: %s, Recived: %s", c.expectedError, err)
-				} else if c.hasError && err.Error() == c.expectedError {
-
-				} else {
+				} else if !c.hasError && err.Error() != c.expectedError {
```
[the remark this answers is not in the document]
Comment (makslion) [reply cmd/getSupportedVersions_test.go:390]: Yep, you are right. Couple of booleans is my doom :laughing:

**Comment by carlkyrillos** [comment]
/lgtm
Reviewed minor change to boolean logic in `cmd/getSupportedVersions_test.go` and we decided that prerequisites documentation will be addressed in a followup ticket.

**Comment by carlkyrillos** [comment]
Followup JIRA: [MGDAPI-3771](https://issues.redhat.com/browse/MGDAPI-3771)

**Pull request merged**